### PR TITLE
Adds API/GPT powered recommendations to listening mode.

### PR DESCRIPTION
**Pull request by andrewroycarter**
# Description of Work

* Listening mode now will call an API that is powered by ChatGPT to generate responses
* Listening mode will fall back to the on device model when the API is disabled remotely
* Minimal error handling is in place - enough for an MVP
    * Translations done by GPT - the language is simple enough that this should be fine
* GPT settings toggle language not translated for now - it's more important / involved so it should be translated by actual people.

# Demo Video

https://github.com/willowtreeapps/vocable-ios/assets/181006/9c2836ac-fe4e-47d7-b8d3-6316494d9855

### CODE REVIEW

File: Vocable/Features/Voice/ListeningResponseContentViewController.swift [hunk 1]
```diff
@@ -30,6 +30,14 @@ final class ListeningResponseContentViewController: PagingCarouselViewController
 
     var disposables = Set<AnyCancellable>()
     var synthesizedSpeechQueue: DispatchQueue!
+    var apiClient: ListenAPIClient?
+    
+    /// When listening mode was used to generate the `Content` displayed, set this to the prompt used to querty the API.
```
Comment (willtellis) [inline Vocable/Features/Voice/ListeningResponseContentViewController.swift]: "query"

**Review by willtellis — APPROVED** [review]

File: Vocable/Supporting Files/ar.lproj/Localizable.strings [hunk 2]
```diff
@@ -115,9 +115,15 @@
 /* Listen mode error state message for when Apple's speech to text service is unavailable */
 "listening_mode.empty_state.speech_unavailable.message" = "تتطلب خدمات الكلام تفعيل إحدى الميزات التالية على جهازك: Siri أو Dictation. يرجى التحقق من إعدادات جهازك وتمكين Siri أو Dictation.";
 
+/* Listen mode error state message for when the vocable API fails */
+"listening_mode.empty_state.vocableAPIFailure.message" = "يرجى التحقق من اتصالك بالإنترنت والمحاولة مرة أخرى في وقت لاحق.";
+
```
Comment (stevefosterwta) [inline Vocable/Supporting Files/ar.lproj/Localizable.strings:120]: Did these come from CrowdIn?

**Review by stevefosterwta — APPROVED** [review]
👍

File: Vocable/Supporting Files/ar.lproj/Localizable.strings (hunk 2 above)
Comment (andrewroycarter) [reply Vocable/Supporting Files/ar.lproj/Localizable.strings:120]: I trusted GPT for these little error strings, but the bigger string that describes the feature I didn't quite trust, that's more important.

Also @stevefosterwta we gotta chat crowdin cause 1) I don't know what it is and how it works and 2) I had to disable the crowdin step on CI as it tries to write to what I assume is a strings repo that now has a protected branch.

**Pull request merged**